### PR TITLE
Limit Pod.List calls by namespace

### LINKED ISSUE #19: Listing all pods in the cluster is too expensive
This triggers our alerts for etcd request latencies and starts effecting other API calls.  Likely not a problem on etcd3, but it's still probably a good idea.

### PR DESCRIPTION
**Pull request by blakebarnett**
On larger clusters with thousands of pods listing all pods is too expensive.

Fixes #19

### CODE REVIEW

**Comment by blakebarnett** [comment]
This should probably move to client initialization time, but works fine otherwise.

File: pkg/goldpinger/k8s.go [hunk 1]
```diff
@@ -16,14 +16,23 @@ package goldpinger
...
-
+	"io/ioutil"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func GetAllPods() map[string]string {
+func getNamespace() string {
+	b, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		log.Println("Unable to determine namespace: ", err.Error())
```
Comment (twexler) [inline pkg/goldpinger/k8s.go]: Just to be explicit here, we should return empty string in this block.
```suggestion
 		log.Println("Unable to determine namespace: ", err.Error())
 		return ""
```

**Comment by blakebarnett** [comment]
sorry about that, I was on vacation :)

I'm not sure how the indentation made it past my vim config (runs go fmt automatically) 🤷‍♂️

File: pkg/goldpinger/k8s.go [hunk 2]
```diff
@@ -16,14 +16,24 @@ package goldpinger
...
 import (
 	"log"
-
+	"io/ioutil"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func GetAllPods() map[string]string {
+func getNamespace() string {
+	b, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```
Comment (dannyk81) [inline pkg/goldpinger/k8s.go]: @blakebarnett @seeker89 sorry for jumping on a merged PR like this ;) but we've hit this on one of our clusters and wanted to share.

Although this is a rare case, a cluster may have Service Account admission controller disabled (I know... I know...) in this case this file doesn't exist.

Could we consider a fallback option to read this from an Env variable? we could pass it on using `metadata.namespace` downward api, similar to how we pass down the hostname.

**Pull request merged**